packages/api-runner/lib/fetchers/getPoolChainData.js

```diff
@@ -409,43 +392,56 @@ const formatPoolChainData = (
     }
 
     // Token listener
-    if (pool.tokenListener.address) {
-      const tokenListenerData = firstBatchValues[pool.tokenListener.address]
-      const tokenFaucetDripAssetAddress = tokenListenerData.asset[0]
-      const tokenListenerData2 =
+    let tokenFaucetDripTokens = []
+    pool.tokenFaucets.forEach((tokenFaucetAddress) => {
+      const tokenFaucetData = firstBatchValues[tokenFaucetAddress]
+      const dripTokenAddress = tokenFaucetData.asset[0]
+      const tokenFaucetData2 =
         secondBatchValues[
-          getExternalErc20AwardBatchName(pool.prizePool.address, tokenFaucetDripAssetAddress)
+          getTokenFaucetDripTokenName(pool.prizePool.address, tokenFaucetAddress, dripTokenAddress)
         ]
-      const tokenFaucetDripToken = {
-        address: tokenFaucetDripAssetAddress.toLowerCase(),
-        amount: formatUnits(tokenListenerData2.balanceOf[0], tokenListenerData2.decimals[0]),
-        amountUnformatted: tokenListenerData2.balanceOf[0],
-        decimals: tokenListenerData2.decimals[0],
-        name: tokenListenerData2.name[0],
-        symbol: tokenListenerData2.symbol[0]
+
+      const dripToken = {
+        address: dripTokenAddress.toLowerCase(),
```

Comment:
Could we add some id so you can link the drip token from this array back to the specific token faucet. Just `tokenFaucetAddress: tokenFaucet.address` would be enough.